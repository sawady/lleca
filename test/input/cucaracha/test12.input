
fun f(h : Int, o : Int, l : Int, a : Int, enter : Int) {
  putChar(h)
  putChar(o)
  putChar(l)
  putChar(a)
  putChar(enter)
}

fun g() {
  f(104, 111, 108, 97, 10)
  f(72, 79, 76, 65, 10)
}

fun main() {
  g()
  g()
}
