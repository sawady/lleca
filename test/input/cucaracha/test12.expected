Cons(
  Function(
    f,
    Cons(
      Parameter(
        h,
        Int
      ),
      Cons(
        Parameter(
          o,
          Int
        ),
        Cons(
          Parameter(
            l,
            Int
          ),
          Cons(
            Parameter(
              a,
              Int
            ),
            Cons(
              Parameter(
                enter,
                Int
              ),
              Nil
            )
          )
        )
      )
    ),
    Unit,
    Cons(
      StmtCall(
        putChar,
        Cons(
          ExprVar(
            h
          ),
          Nil
        )
      ),
      Cons(
        StmtCall(
          putChar,
          Cons(
            ExprVar(
              o
            ),
            Nil
          )
        ),
        Cons(
          StmtCall(
            putChar,
            Cons(
              ExprVar(
                l
              ),
              Nil
            )
          ),
          Cons(
            StmtCall(
              putChar,
              Cons(
                ExprVar(
                  a
                ),
                Nil
              )
            ),
            Cons(
              StmtCall(
                putChar,
                Cons(
                  ExprVar(
                    enter
                  ),
                  Nil
                )
              ),
              Nil
            )
          )
        )
      )
    )
  ),
  Cons(
    Function(
      g,
      Nil,
      Unit,
      Cons(
        StmtCall(
          f,
          Cons(
            ExprConstNum(
              104
            ),
            Cons(
              ExprConstNum(
                111
              ),
              Cons(
                ExprConstNum(
                  108
                ),
                Cons(
                  ExprConstNum(
                    97
                  ),
                  Cons(
                    ExprConstNum(
                      10
                    ),
                    Nil
                  )
                )
              )
            )
          )
        ),
        Cons(
          StmtCall(
            f,
            Cons(
              ExprConstNum(
                72
              ),
              Cons(
                ExprConstNum(
                  79
                ),
                Cons(
                  ExprConstNum(
                    76
                  ),
                  Cons(
                    ExprConstNum(
                      65
                    ),
                    Cons(
                      ExprConstNum(
                        10
                      ),
                      Nil
                    )
                  )
                )
              )
            )
          ),
          Nil
        )
      )
    ),
    Cons(
      Function(
        main,
        Nil,
        Unit,
        Cons(
          StmtCall(
            g,
            Nil
          ),
          Cons(
            StmtCall(
              g,
              Nil
            ),
            Nil
          )
        )
      ),
      Nil
    )
  )
)
